hugr/src/hugr/serialize.rs

```diff
@@ -26,16 +26,21 @@ use super::{HugrMut, HugrView};
 /// will try to deserialize them in order.
 #[derive(Serialize, Deserialize)]
 #[serde(tag = "version", rename_all = "lowercase")]
-enum Versioned {
+enum Versioned<T> {
```

Comment:
why is this generic? it implies a new generic parameter will need to be added for each version of the payload.

Reply:
From below it looks like this is generic to allow versioned payloads for different types. Maybe call the type parameter `V1` for clarity.